Mathlib/Topology/ContinuousFunction/UniqueCFC.lean

```diff
@@ -196,3 +196,211 @@ instance NNReal.instUniqueContinuousFunctionalCalculus [UniqueContinuousFunction
+    [TopologicalSpace A] [T2Space A] [NonUnitalRing A] [StarRing A] [Module 𝕜 A]
+    [IsScalarTower 𝕜 A A] [SMulCommClass 𝕜 A A] [h : ∀ a : A, CompactSpace (quasispectrum 𝕜 a)] :
+    UniqueNonUnitalContinuousFunctionalCalculus 𝕜 A where
+  eq_of_continuous_of_map_id s hs _inst h0 φ ψ hφ hψ h := by
+    rw [DFunLike.ext'_iff, ← Set.eqOn_univ, ← (ContinuousMapZero.adjoin_id_dense h0).closure_eq]
+    refine Set.EqOn.closure (fun f hf ↦ ?_) hφ hψ
+    rw [← NonUnitalStarAlgHom.mem_equalizer]
+    apply adjoin_le ?_ hf
+    rw [Set.singleton_subset_iff]
+    exact h
+  compactSpace_quasispectrum := h
+
+instance RCLike.instUniqueNonUnitalContinuousFunctionalCalculus [NonUnitalNormedRing A]
+    [StarRing A] [CompleteSpace A] [NormedSpace 𝕜 A] [IsScalarTower 𝕜 A A] [SMulCommClass 𝕜 A A] :
+    UniqueNonUnitalContinuousFunctionalCalculus 𝕜 A :=
+  RCLike.uniqueNonUnitalContinuousFunctionalCalculus_of_compactSpace_quasispectrum
+
+end RCLike
+
+section NNReal
+open NNReal
+
+variable {X : Type*} [TopologicalSpace X] [Zero X]
+
+namespace ContinuousMapZero
+
+/-- This map sends `f : C(X, ℝ)` to `Real.toNNReal ∘ f`, bundled as a continuous map `C(X, ℝ≥0)`. -/
+noncomputable def toNNReal (f : C(X, ℝ)₀) : C(X, ℝ≥0)₀ := ⟨.realToNNReal |>.comp f, by simp⟩
+
+@[simp]
+lemma toNNReal_apply (f : C(X, ℝ)₀) (x : X) : f.toNNReal x = Real.toNNReal (f x) := rfl
+
+@[fun_prop]
+lemma continuous_toNNReal : Continuous (toNNReal (X := X)) := by
+  rw [continuous_induced_rng]
+  convert_to Continuous (ContinuousMap.toNNReal ∘ ((↑) : C(X, ℝ)₀ → C(X, ℝ))) using 1
+  exact ContinuousMap.continuous_comp _ |>.comp continuous_induced_dom
+
+lemma toContinuousMapHom_toNNReal (f : C(X, ℝ)₀) :
+    (toContinuousMapHom (X := X) (R := ℝ) f).toNNReal =
+      toContinuousMapHom (X := X) (R := ℝ≥0) f.toNNReal :=
+  rfl
+
+@[simp]
+lemma toNNReal_smul (r : ℝ≥0) (f : C(X, ℝ)₀) : (r • f).toNNReal = r • f.toNNReal := by
+  ext x
+  by_cases h : 0 ≤ f x
+  · simpa [max_eq_left h, NNReal.smul_def] using mul_nonneg r.coe_nonneg h
+  · push_neg at h
+    simpa [max_eq_right h.le, NNReal.smul_def]
+      using mul_nonpos_of_nonneg_of_nonpos r.coe_nonneg h.le
+
+@[simp]
+lemma toNNReal_neg_smul (r : ℝ≥0) (f : C(X, ℝ)₀) : (-(r • f)).toNNReal = r • (-f).toNNReal := by
+  rw [NNReal.smul_def, ← smul_neg, ← NNReal.smul_def, toNNReal_smul]
+
+-- Without this, Lean fails to find an instance of
+-- `AddHomClass (C(X, ℝ≥0)₀ →⋆ₙₐ[ℝ≥0] C(X, ℝ≥0)) C(X, ℝ≥0)₀ C(X, ℝ≥0)`.
+set_option synthInstance.maxHeartbeats 30000 in
+lemma toNNReal_mul_add_neg_mul_add_mul_neg_eq (f g : C(X, ℝ)₀) :
+    ((f * g).toNNReal + (-f).toNNReal * g.toNNReal + f.toNNReal * (-g).toNNReal) =
+    ((-(f * g)).toNNReal + f.toNNReal * g.toNNReal + (-f).toNNReal * (-g).toNNReal) := by
+  apply toContinuousMap_injective
+  simpa only [← toContinuousMapHom_apply, map_add, map_mul, map_neg, toContinuousMapHom_toNNReal]
+    using (f : C(X, ℝ)).toNNReal_mul_add_neg_mul_add_mul_neg_eq g
```

Comment:
```suggestion
-- Without this, Lean fails to find an instance of
-- `AddHomClass (C(X, ℝ≥0)₀ →⋆ₙₐ[ℝ≥0] C(X, ℝ≥0)) C(X, ℝ≥0)₀ C(X, ℝ≥0)`.
lemma toNNReal_mul_add_neg_mul_add_mul_neg_eq (f g : C(X, ℝ)₀) :
    ((f * g).toNNReal + (-f).toNNReal * g.toNNReal + f.toNNReal * (-g).toNNReal) =
    ((-(f * g)).toNNReal + f.toNNReal * g.toNNReal + (-f).toNNReal * (-g).toNNReal) := by
  have : AddHomClass (C(X, ℝ≥0)₀ →⋆ₙₐ[ℝ≥0] C(X, ℝ≥0)) C(X, ℝ≥0)₀ C(X, ℝ≥0) := 
    SemilinearMapClass.toAddHomClass
  apply toContinuousMap_injective
  simpa only [← toContinuousMapHom_apply, map_add, map_mul, map_neg, toContinuousMapHom_toNNReal]
    using (f : C(X, ℝ)).toNNReal_mul_add_neg_mul_add_mul_neg_eq g
```
Still ugly, but less so than the increasing the timeout. Also works for the next one.

Reply:
gross, but okay.